File: java设计模式/src/组合模式/Composite.java
package 组合模式;

import java.util.ArrayList;

public class Composite extends Component {

    ArrayList<Component> arrayList = new ArrayList<> ();

    public Composite(String name) {
        super ( name );
    }


    @Override
    public void addComponent(Component component) {
        this.arrayList.add ( component );

    }

    @Override
    public void remove(Component component) {
        this.arrayList.remove ( component );
    }


    @Override
    public void display() {
        System.out.println ( this.name );
        for (Component component : this.arrayList) {
            component.display ();
        }

    }
}

File: 9.组合模式/geqiangbao_20201127_Component_swift/geqiangbao_20201127_Component_swift/案例一：原始案例/Composite.swift
//
//  Composite.swift
//  geqiangbao_20201127_Component_swift
//
//  Created by xiaobao on 2020/11/27.
//

import UIKit


//具体子节点
class Composite: NSObject,ComponentProtocol {
    var name: String = ""
    
    var components: Array<ComponentProtocol> = []
    
    init(name: String) {
        self.name = name
    }
    
    func doSomething() {
        print("q345")
    }
//   添加
    func addComponent(child:ComponentProtocol){
        self.components.append(child)
    }

//   删除
    func removeCoponent(child:ComponentProtocol){
        var iterator = self.components.makeIterator()
        var model:ComponentProtocol?
        var index = -1
        while true {
            model = iterator.next()
            if model == nil {
                break
            }
            index += 1
            if model?.name == child.name {
                self.components.remove(at: index)
                break
            }
        }
    }
//    获取
    func get(index:Int) -> ComponentProtocol?{
        if index < self.components.count {
            return self.components[index]
        }
        return nil
    }
    
    
    /// 清空
    func clear(){
        self.components.removeAll()
    }


    
    

}

//NSString *priceTick = [[QDSettingManager shareInstance]getPriceTick_exchangeId:model.exchangeId instrumentId:model.instrumentId];

